src/lib/arch/riscv/Instruction_decode.cc

```diff
@@ -310,6 +310,33 @@ void Instruction::decode() {
       knownOffset_ = sourceImm_;
       break;
   }
+
+  // Calculate the instruction's group based on identifiers
+  // Set base group
+  uint16_t group = InstructionGroups::INT;
+  if (isInstruction(InsnType::isFloat)) group = InstructionGroups::FLOAT;
+  // Identify subgroup type
+  if (isInstruction(InsnType::isBranch))
+    group = InstructionGroups::BRANCH;
+  else if (isInstruction(InsnType::isLoad))
+    group += 8;
+  else if (isInstruction(InsnType::isStore))
+    group += 9;
+  else if (isInstruction(InsnType::isDivide))
+    group += 7;
+  else if (isInstruction(InsnType::isMultiply))
+    group += 6;
+  else if (isInstruction(InsnType::isShift) ||
+           isInstruction(InsnType::isConvert))
+    group += 5;
+  else if (isInstruction(InsnType::isLogical))
+    group += 4;
+  else if (isInstruction(InsnType::isCompare))
+    group += 3;
+  else
+    group += 2;  // Default return is {Data type}_SIMPLE_ARTH
+
```

Comment:
Would a switch statement on `instructionIdentifier` not be viable? Would be slightly nicer to read

Reply:
I don't this a switch would work here, or would decrease readability if it did. `instructionIdentifier` is just an `uint32_t` and so the associated value used in the switch has no meaning. With the current implementation the `if` clause contains what is being checked against (i.e. `isMultiply`)

Reply:
I hadn't considered the case where an instruction has multiple types so I think this is probably the simpler solution (although not pretty). No action required